ui/app/rif/components/chainAddresses.js

```diff
@@ -128,10 +127,7 @@ class ChainAddresses extends Component {
     this.setState({ insertedAddress: address });
   }
 
- timeoutToRedirect = (selectedChainAddress, isRetry = true) => setTimeout(async () => {
-    if (!isRetry) {
-     await this.props.deletePendingChainAddress(selectedChainAddress, !!this.props.subdomainName);
-    }
+ timeoutToRedirect = (selectedChainAddress) => setTimeout(async () => {
```

Comment:
This timeout should be saved on the component object and cleaned out on the componentWillUnmount hook to avoid having problems of re-route later.
Something like this should work:
```
constructor(props) {
    super(props);
    this.timeouts = [];
}
...
timeoutToRedirect = ....
this.timeouts.push(timeoutToRedirect);
...

componentWillUnmount () {
    this.timeouts.forEach(timeout => clearTimeout(timeout));
}
```

Reply:
There i implemented that!